### PR TITLE
[TTAHUB-1782] Add new permissions, feature flag, and AR training page

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

This PR starts to lay the foundation for the new 'Training reports page' frontend. It adds a feature flag for training reports as well as read and write permission for the new report type.

**Note:** If your user has ADMIN you will see the Training reports NAV regardless of feature flag settings.

**New Flag/Permissions:**
image

**New Page:**
image

## How to test

1. Review the code
2. For a non-admin user make sure you can set/change the feature flag for training reports.
3. A non-admin user without the feature flag should NOT be able to go to the new page.
4. A non-admin user with the feature flag should be able to visit the page.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1782

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

**Review by kryswisnaskas — COMMENTED** [review]
Looks good! 👍 

Only one comment about the down migration. 🙇

File: src/migrations/20230515094028-training-reports.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        `SELECT
+                set_config('audit.loggedUser', '${loggedUser}', TRUE) as "loggedUser",
+                set_config('audit.transactionId', NULL, TRUE) as "transactionId",
+                set_config('audit.sessionSig', '${sessionSig}', TRUE) as "sessionSig",
+                set_config('audit.auditDescriptor', '${auditDescriptor}', TRUE) as "auditDescriptor";`,
+        { transaction },
+      );
+      await queryInterface.sequelize.query('DELETE FROM "Permissions" WHERE "scopeId" IN (7, 8);', { transaction });
+      await queryInterface.sequelize.query('DELETE FROM "Scopes" WHERE id IN (7, 8);', { transaction });
+    });
```
[the remark this answers is not in the document]
Comment (AdamAdHocTeam) [reply src/migrations/20230515094028-training-reports.js:64]: It doesn't look like removing ENUM values is supported. This is probably why we have never had it in the down path before. Else we would need to completely delete it and recreated the ENUM.

[the remark this answers is not in the document]
Comment (AdamAdHocTeam) [reply src/migrations/20230515094028-training-reports.js:64]: https://www.postgresql.org/docs/current/datatype-enum.html

**Review by kryswisnaskas — APPROVED** [review]

File: src/migrations/20230515094028-training-reports.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvms) [reply src/migrations/20230515094028-training-reports.js:64]: you're right, removing values from an enum isn't directly supported, but you can work around it by creating a new enum and replacing the old one with it. it's a bit clunky, but it's possible.

I've done it in the `down` for feature flag changes, e.g.:

https://github.com/HHS/Head-Start-TTADP/blob/987c0489c2e689809c327efc9a91530aa7500031/src/migrations/20230110204839-remove-goals-objectives-feature-flag.js#L18-L35

[the remark this answers is not in the document]
Comment (nvms) [reply src/migrations/20230515094028-training-reports.js:64]: here's another example that I think is a little more readable:

https://github.com/HHS/Head-Start-TTADP/blob/987c0489c2e689809c327efc9a91530aa7500031/src/migrations/20221201231306-user-settings-ps-emails.js#L52-L66

**Pull request merged**